client/engine/messageservice/p2p-message-service/service.go

```diff
@@ -59,12 +62,15 @@ func (ms *P2PMessageService) Id() peer.ID {
 }
 
 // NewMessageService returns a running P2PMessageService listening on the given ip, port and message key.
-func NewMessageService(ip string, port int, me types.Address, pk []byte) *P2PMessageService {
+func NewMessageService(ip string, port int, me types.Address, pk []byte, logWriter io.Writer) *P2PMessageService {
+	logging.ConfigureZeroLogger()
+
 	ms := &P2PMessageService{
 		toEngine:    make(chan protocols.Message, BUFFER_SIZE),
 		newPeerInfo: make(chan basicPeerInfo, BUFFER_SIZE),
 		peers:       &safesync.Map[basicPeerInfo]{},
 		me:          me,
+		logger:      zerolog.New(logWriter).With().Timestamp().Str("message-service", me.String()[0:8]).Caller().Logger(),
```

Comment:
Possibly beyond the scope of this PR but it could be nice to have a helper function that returns the shortened address.